packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -894,6 +894,29 @@ const handleCfnFiles = (eventMap: EventMap, category: string, resource: string,
   });
 };
 
+const getNestedStackParameters = async (context: $TSContext, categories: Array<string>, amplifyMeta: any) => {
+  const s3 = await S3.getInstance(context);
+  const currentEnvName = context.amplify.getEnvInfo().envName;
+  const urlParametersMap = {};
+  for (const category of categories) {
+    const resources = Object.keys(amplifyMeta[category]);
+    for (const resource of resources) {
+      const resourceDetails = amplifyMeta[category][resource];
+      if (resourceDetails.providerMetadata) {
+        const templateURL = resourceDetails.providerMetadata.s3TemplateURL;
+        try {
+          const key = templateURL.split(resourceDetails.s3Bucket.deploymentBucketName + '/')[1];
+          const jsonBody = await s3.getFile({ Key: key }, currentEnvName);
```

Comment:
why do we need to go to S3 for this? The stacks that were just uploaded should all be present in the `backend/<category>/<resource name>/build` directory for each resource. Better yet would be to put this check before we even upload the generated stacks to S3. Although this might get complicated for AppSync because it has some internal upload logic of it's own